shell/actions/order_assign-guestorders2customer.php

```diff
@@ -0,0 +1,83 @@
+<?php 
+
+require_once 'public_html/app/Mage.php';
+umask(0); 
+Mage::init();
+
+function getInput($minLength = '1')
+{
+	$handle = fopen("php://stdin", "r");
+	$line = fgets($handle);
+	if (strlen(trim($line)) < $minLength) {
+            echo "error - ABORTING!\n";
+	    exit;
+	}	
+	fclose($handle);
+
+	return trim($line);
+}
+
+function getStoreViewList()
+{
+    $_store = array();
+    foreach (Mage::app()->getWebsites() as $website) {
+        foreach ($website->getGroups() as $group) {
+            $stores = $group->getStores();
+            foreach ($stores as $store) {
+                $_store[] = array("id" => $store->getWebsiteId(), "name" => $store->getName());
+            }
+        }
+    }
+	
+    return $_store;
+}
+
+// Print stores list
+$allStoresArray = getStoreViewList();
+foreach($allStoresArray as $store)
+{
+	echo "[". $store['id'] . "] " . $store['name'] . "\r\n";
+}
+
+
+echo "Please input Website ID: ";
+$websiteID = getInput();
+
+echo "Please input customer email: ";
+$customerEmailID = getInput('5');
+
+
+/****************************************/
+
+if(isset($customerEmailID) && isset($websiteID))
+{
+	$toCustomer = Mage::getModel('customer/customer')
+		->setWebsiteId($websiteID)
+		->loadByEmail($customerEmailID);
+
+
+	$orders = Mage::getModel('sales/order')->getCollection()
+		->addAttributeToFilter('store_id',$websiteID)
+		->addAttributeToFilter('customer_is_guest', array('nin' => array(0)))
+		->addAttributeToFilter('customer_email', $customerEmailID);
+
+		
+	foreach($orders as $order)
+	{
+		try {
+			$orderByID = Mage::getModel('sales/order')->loadByIncrementId($order['increment_id']);
```

Comment:
Why do you need to reload order? From reading the code, it seems to me that `$order` should do just fine. No?

Also, can you also do a similar cleanup on tabs (use 4 spaces), and empty lines (e.g. 57, 58)

Reply:
its needed because a model loaded over a collection often misses data, causing weird bugs when one tries to save them

Reply:
In that case, we can return `getAllIds()` from the collection. Added suggestion.

Reply:
`addAttributesToSelect('*')` does not work to avoid missing data?

However ... there is another way to update data, that is probably faster ...

https://github.com/OpenMage/magento-lts/blob/dac0eee6f6bde6345274e7a6143481d596ba5307/app/code/core/Mage/Sales/Model/Resource/Order/Abstract.php#L312